src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -178,9 +178,11 @@ void AdaptDispatch::_WriteToBuffer(const std::wstring_view string)
 
     _ApplyCursorMovementFlags(cursor);
 
-    // Notify UIA of new text.
+    // Notify terminal and UIA of new text.
     // It's important to do this here instead of in TextBuffer, because here you
-    // have access to the entire line of text.
+    // have access to the entire line of text, whereas TextBuffer writes it one
+    // character at a time via the OutputCellIterator.
+    _api.NotifyTextLayoutUpdated();
```

Comment:
this sounds expensive; is it?

Reply:
> We could raise the event only once after processing an entire VT string to reduce the cost.

That makes sense. We don't render in between parsing so doing it at every write step vs at the end of the entire vt string is equivalent.

I'm also planning to introduce throttling in this codepath, but I've a few queries like currently typing acts like a way for user to dismiss highlights from the screen. Once we introduce throttling this won't happen anymore. Instead typing into the terminal will dispatch a delayed refresh of search highlights which will bring back the highlights after a short delay. So, I'm not sure how a user would dismiss highlights in this case.

Any suggestions?

Reply:
Hmm, I may be misunderstanding you, but what you want is to simply remove the search highlights when someone types, right? That's basically just like `TrySnapOnInput` - if you search it in the project, there should only be a few hits. That function is what makes sure you're scrolled all the way down when you type, and we could just extend that, or add code near it, to also dismiss the highlights.

Reply:
> what you want is to simply remove the search highlights when someone types, right?
Huh.. quite the opposite 😅 Highlights shouldn't be removed if search box is open. So, unless the searchbox is closed, we should keep the results fresh and active.

To be honest, I don't know what is a good UX here 😄

Reply:
Oh I see, now I got it. 🙂
In a perfect world we'd probably reflow the highlights while reflowing the text during a resize, but I don't think that's worth the effort at the moment. I think putting it into the throttled `updatePatternLocations` callback is probably the right choice for now as it presents a decent trade-off between correctness and complexity.

Reply:
FYI I just started making the necessary changes. 😊